logcheck/pkg/logcheck.go

```diff
@@ -496,3 +508,37 @@ func checkForIfEnabled(i *ast.IfStmt, pass *analysis.Pass, c *config) {
 			funcCall, varName, funcCall, varName, varName),
 	})
 }
+
+func checkForVerboseZero(expr *ast.CallExpr, pass *analysis.Pass) {
+	if typeAndValue, ok := pass.TypesInfo.Types[expr]; ok {
+		switch t := typeAndValue.Type.(type) {
+		case *types.Named:
+			if typeName := t.Obj(); typeName != nil {
+				if pkg := typeName.Pkg(); pkg != nil {
+					if typeName.Name() == "Verbose" && pkg.Path() == "k8s.io/klog/v2" {
+						lit, ok := expr.Args[0].(*ast.BasicLit)
+						if !ok || lit.Value != "0" {
+							return
+						}
+						msg := fmt.Sprintf("klog does not allow verbose zero")
+						pass.Report(analysis.Diagnostic{
+							Pos:     expr.Fun.Pos(),
+							Message: msg,
+						})
+					}
+					if typeName.Name() == "Logger" && pkg.Path() == "github.com/go-logr/logr" {
+						lit, ok := expr.Args[0].(*ast.BasicLit)
+						if !ok || lit.Value != "0" {
+							return
+						}
+						msg := fmt.Sprintf("logger does not allow verbose zero")
```

Comment:
I think we should explain the rationale for the warning differently: it is allowed, just not useful.

```suggestion
						msg := fmt.Sprintf("Logging with V(0) is semantically equivalent to the same expression without it and just causes unnecessary overhead. It should get removed.")
```